packages/hash/api/src/db/postgres/util.test.ts

```diff
@@ -0,0 +1,70 @@
+import { transaction } from "./util";
+import { PoolConnection, TransactionConnection } from "./types";
+
+describe("transaction flattening", () => {
+  it("allows a single top-level transaction", async () => {
+    const transactionsHandler = jest.fn().mockResolvedValue("ok");
+    const transactionConn = {
+      // The transaction connection is untagged when it is created by the DB
+      // the type field will be present after creating a transaction.
+      ...({} as TransactionConnection),
```

Comment:
```suggestion
      ...({ type: "Transaction" } as TransactionConnection),
```

Reply:
This is left out intentionally to check if the connection is properly tagged when a transaction is requested from the poolconneciton

Reply:
Ah apologies, the type confused me but I see what's going on now.